### DISCUSSION TITLE
Can livewire be used without laravel?

### DISCUSSION THREAD

**Discussion by mabasic** [livewire/livewire#4024] [3. Feature ideas]
I have recently started using livewire and I am really liking it. It removes all JS witchcraft and lets me work with PHP only. If needed I can easily use alpine.js again without writing js, and tailwind to avoid writing css.

My question is can livewire be used without laravel? And if no, can it be made in a way that it is not built upon laravel but plain php somehow. 

Don't know if this is even possible, just thinking out loud.

**Comment by joshhanley** [comment]
@mabasic short answer is no.

My question for you would be why. Why would you want to use it without Laravel?

Hope this helps!

**Reply by mabasic** [reply]
I don't know why exactly 😊 

It seems like a good way to write client side interactivity with php. And if it could be used with other frameworks/plain php aka framework agnostic I think that it could be a huge asset to the whole php ecosystem.

**Reply by joshhanley** [reply]
@mabasic fair enough 🙂  wasn't sure if you had a specific need or whether it was more like a thought experiment.

So interestingly enough, Livewire's front end (javascript assets) are actually being used in other projects, I know of one called cbwire which is made by @grantcopley for ColdFusion. Have a look here #3568 for a discussion on that.

In terms of making it generic for PHP though, currently that would require a near full rewrite of the backend as it was written from the ground up as a Laravel framework package, so it heavily relies on things like service providers, collections, string helper methods, etc.

So I think it would be pretty tough/ near impossible to adapt the current backend for generic PHP use.

But that's not to say it can't or shouldn't be done 🙂
